assets/dev/js/editor/elements/views/base.js

```diff
@@ -609,17 +617,152 @@ BaseElementView = BaseContainer.extend( {
+				return true;
 			}
 
-			if ( ! isContentChanged ) {
+			if ( render && ! isContentChanged ) {
 				this.renderUI();
-				return;
+
+				return true;
 			}
 		}
 
-		// Re-render the template
-		this.renderHTML();
+		return false;
+	},
+
+	/**
+	 * Function linksData().
+	 *
+	 * Link data to allow partial render, instead of full re-render
+	 *
+	 * How to use?
+	 *  If the element which should be rendered for a setting key is known in advance, it's possible to add the following attributes to the element to avoid full re-render:
+	 *  Example for repeater item:
+	 * 'data-link-type': 'repeater-item',      // Type of link (to know how to behave)
+	 * 'data-link-setting': 'tab_title',       // Setting key that effect the link.
+	 * 'data-link-index': tabCount,            // Index is required for repeater items.
+	 *
+	 * Example for content:
+	 * 'data-link-type': 'content',
+	 * 'data-link-setting': 'testimonial_content',
+	 * By adding the following example attributes inside the widget,
+	 * to element that contains the attributes, it will effect his innerHTML and it will be linked to the 'testimonial_content'  setting value.
+	 *
+	 * Current Limitation:
+	 * Not working with dynamics, will required full re-render.
+	 */
+	linksData() {
+		/**
+		 * @type {Array.<DataLink>}
+		 */
+		this.links = [];
+
+		const id = this.$el.data( 'id' );
+
+		if ( ! id ) {
+			return;
+		}
+
+		const $dataLink = this.$el.find( '[data-link-type]' );
+
+		if ( ! $dataLink.length ) {
+			return;
+		}
+
+		const self = this;
+
+		$dataLink.filter( function() {
+			const $current = jQuery( this );
+
+			// To support nested links bypass nested links that are not part of the current.
+			if ( $current.closest( '.elementor-element' ).data( 'id' ) === id ) {
+				if ( this.dataset.linkType ) {
+					self.links.push( {
+						el: this,
+						dataset: this.dataset,
+					} );
+				}
+			}
+		} );
+	},
+
+	/**
+	 * Function renderLinks().
+	 *
+	 * Render linked data.
+	 *
+	 * @param {Object} settings
+	 * @param {Array.<DataLink>} links
+	 * @returns {boolean} - false on fail.
+	 */
+	renderLinks( settings, links ) {
+		let changed = false;
+
+		for ( const link of links ) {
+			let leave = false;
+
+			switch ( link.dataset.linkType ) {
+				case 'repeater-item': {
+					const container = Object.values( this.container.repeaters )[ 0 ].children.find(
+						( i ) => i.id === settings.attributes._id
+					);
+
+					if ( ( container?.parent?.children.indexOf( container ) + 1 ) === parseInt( link.dataset.linkIndex ) ) {
+						const change = settings.changed[ link.dataset.linkSetting ];
+
+						if ( change ) {
+							changed = true;
+							jQuery( link.el ).html( change );
+							leave = true;
+						}
+					}
+				}
+				break;
+
+				case 'content': {
+					if ( undefined !== settings.changed[ link.dataset.linkSetting ] ) {
+						const change = settings.changed[ link.dataset.linkSetting ];
+
+						changed = true;
+
+						link.el.innerHTML = change;
+					}
```

Comment:
```suggestion
					if ( undefined !== settings.changed[ link.dataset.linkSetting ] ) {
						const change = settings.changed[ link.dataset.linkSetting ];

						changed = true;

						link.el.innerHTML = change;
					}
```
If change found and changed break the loop